docs/overview/10-tutorials/01-nextjs-circom.md

```diff
@@ -77,61 +77,126 @@ Now let's create a backend API in our NextJS application which will take the pro
+      await registerVk();
+      await new Promise((resolve) => setTimeout(resolve, 5000));
+    }
 
-        if(requestResponse.data.optimisticVerify != "success"){
-            console.error("Proof verification, check proof artifacts");
-            return;
-        }
+    const vk = fs.readFileSync(
+      path.join(
+        process.cwd(),
+        "public",
+        "assets",
+        "vkey.json"
+      ),
+      "utf-8"
+    );
+
+    const params = {
+      proofType: "groth16",
+      vkRegistered: true,
+      proofOptions: {
+        library: "snarkjs",
+        curve: "bn128",
+      },
+      proofData: {
+        proof: req.body.proof,
+        publicSignals: req.body.publicInputs,
+        vk: JSON.parse(vk).vkHash || JSON.parse(vk).meta.vkHash,
+      },
+    };
+
+    const requestResponse = await axios.post(
+      `${API_URL}/submit-proof/${process.env.API_KEY}`,
+      params
+    );
+    console.log(requestResponse.data);
+
+    if (requestResponse.data.optimisticVerify != "success") {
+      console.error("Proof verification, check proof artifacts");
+      return;
+    }
 
-        while(true){
-            const jobStatusResponse = await axios.get(`${API_URL}/job-status/${process.env.API_KEY}/${requestResponse.data.jobId}`);
-            if(jobStatusResponse.data.status === "IncludedInBlock"){
-                console.log("Job Included in Block successfully");
-                res.status(200).json(jobStatusResponse.data);
-                return;
-            }else{
-                console.log("Job status: ", jobStatusResponse.data.status);
-                console.log("Waiting for job to finalize...");
-                await new Promise(resolve => setTimeout(resolve, 5000)); // Wait for 5 seconds before checking again
-            }
+    while (true) {
+      try{
+        const jobStatusResponse = await axios.get(
+          `${API_URL}/job-status/${process.env.API_KEY}/${requestResponse.data.jobId}`
+        );
+        if (jobStatusResponse.data.status === "IncludedInBlock") {
+          console.log("Job Included in Block successfully");
+          res.status(200).json(jobStatusResponse.data);
+          return;
+        } else {
+          console.log("Job status: ", jobStatusResponse.data.status);
+          console.log("Waiting for job to finalize...");
+          await new Promise((resolve) => setTimeout(resolve, 5000)); // Wait for 5 seconds before checking again
+        }
+      }catch (error: any) {
+        if (error.response && error.response.status === 503) {
+          console.log("Service Unavailable, retrying...");
+          await new Promise((resolve) => setTimeout(resolve, 5000)); // Wait for 5 seconds before retrying
         }
-    }catch(error){
-        console.log(error)
+      }
     }
+  } catch (error) {
+    console.log(error);
+  }
+}
+
+async function registerVk() {
+
+  const API_URL = "https://relayer-api.horizenlabs.io/api/v1";
```

Comment:
You could pull this out constant out and define it at the top of this file. Both `handler` and `registerVk` use it